src/operator/subgraph/dnnl/dnnl_transformer.cc

```diff
@@ -86,8 +111,21 @@ static bool SgDNNLSelfAttQKInferType(const nnvm::NodeAttrs& attrs,
         << "QuantizedSelfAttentionQK only supports int8 input, while " << in_types->at(0)
         << " is given.";
 
-    TYPE_ASSIGN_CHECK(*in_types, 1, mshadow::kFloat32);
-    TYPE_ASSIGN_CHECK(*in_types, 2, mshadow::kFloat32);
+    if constexpr (qk_mode == qk_common::mode::without_split) {
+      if (in_types->at(1) == mshadow::kBfloat16) {
+        return false;
+      }
```

Comment:
This check is redundant as (if the program gets to this point) we have already checked in line 100 that in_type0 == in_type1 and in line 106 that in_type0 != bfloat16.

Reply:
done